src/v/cluster/rm_stm.h

```diff
@@ -241,6 +241,8 @@ class rm_stm final : public persisted_stm {
         model::offset lso_bound;
         std::optional<expiration_info> info;
 
+        std::optional<int32_t> seq;
```

Comment:
do we have a named type for `tx_seq` ?

Reply:
Yes.
Out tx_seq is int64_t named type
We need to return int32_t. It is can not convert easy from optional<model::tx_seq> to int32_t in value_or(-1)

Reply:
actually `tx_seq` is an id of a transaction within a session while `seq` is a sequence number of a batch; they represent different things

Reply:
so should we have a named_type for this int32_t batch sequence ?

Reply:
In general I think it's a good idea but it looks like a ton of work to include it to this PR - we need to change `record_batch_header` but a lot of things depends on it. @VadimPlh can you please create an issue and add a TODO comment pointing to the issue.

Reply:
https://github.com/redpanda-data/redpanda/issues/8128